packages/react/src/box/styles.ts

```diff
@@ -387,12 +387,18 @@ export const linkStyles = {
 type FocusProps = Partial<{
 	/** If true, the focus indicator will be shown when the element receives keyboard focus. */
 	focus: boolean;
+	/** If true, the focus indicator will be shown when the element receives keyboard and programmatic focus. */
+	focusAll: boolean;
```

Comment:
This feels weird to me. It feels like `focus` or `focusAll` should not both be able to be set at the same time. What if `focus` accepted a `boolean` or an "all" string? Is that weird?

Reply:
Yeah, that's an option. I don't love `focusAll` but it's hard given the `focus` starting point. I also don't love the obfuscated values that you can pass super props. Let me digest this for a bit. I think I like where you're going with this, though.

Reply:
I don't like `focus` either, because to me it sounds like it will received `focus`. Perhaps we could deprecate this and use a new `focusFor` `'all' | 'keyboard'` prop?

Reply:
Yeah, I think that's the way to go. 

- [ ] `focusFor` prop, deprecate `focus`